File: server/src/main/java/com/site/forum/service/CommentService.java
package com.site.forum.service;

import com.site.forum.entity.Comment;
import com.site.forum.entity.Post;
import com.site.forum.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CommentService {

    Comment create(Comment comment, Post post);
    void delete(Long id);
    Comment getById(Long id);
    Page<Comment> getByPostId(Long postId, Pageable pageable);
    Page<Comment> getByUserUsername(String username, Pageable pageable);
    void like(Comment comment, User user);
    void dislike(Comment comment, User user);
}

File: client/src/services/CommentService.js
import axios from "../api/axios"

import { POST_API_URL } from "../utils/urls"

export const CommentService = {
    async create(postId, payload) {
        const {data} = await axios.post(`${POST_API_URL}/${postId}/comments`, payload)
        return data
    },

    async like(commentId, payload) {
        const {data} = await axios.post(`${POST_API_URL}/comments/${commentId}/like`, payload)
        return data
    },

    async dislike(commentId, payload) {
        const {data} = await axios.post(`${POST_API_URL}/comments/${commentId}/dislike`, payload)
        return data
    }
}